Packs/IntSight/Integrations/IntSight/IntSight.py

```diff
@@ -701,21 +703,181 @@ def search_for_ioc():
+        url_info['AntivirusDetectionRatio'] = demisto.get(ioc_data, 'Data.AntivirusDetectionRatio')
+        url_info['AntivirusDetections'] = demisto.get(ioc_data, 'Data.AntivirusDetections')
+        url_info['AntivirusScanDate'] = demisto.get(ioc_data, 'Data.AntivirusScanDate')
+        url_info['RelatedHashes'] = {
+            'communicating': demisto.get(ioc_data, 'Data.RelatedHashes.communicating'),
+            'downloaded': demisto.get(ioc_data, 'Data.RelatedHashes.downloaded'),
+            'referencing': demisto.get(ioc_data, 'Data.RelatedHashes.referencing'),
+        }
+
+    hash_info = {}
+    if ioc_context['Type'] == 'Hashes':
+        hash_info['Name'] = ioc_context['Value']
+        hash_info[hash_identifier(ioc_context['Value'])] = ioc_context['Value']
+        if translate_severity(ioc_readable['Severity']) == 3:
+            hash_info['Malicious'] = malicious_dict
+
+        hash_info['AntivirusDetectedEngines'] = demisto.get(ioc_data, 'Data.AntivirusDetectedEngines')
+        hash_info['AntivirusDetectionRatio'] = demisto.get(ioc_data, 'Data.AntivirusDetectionRatio')
+        hash_info['AntivirusDetections'] = demisto.get(ioc_data, 'Data.AntivirusDetections')
+        hash_info['AntivirusScanDate'] = demisto.get(ioc_data, 'Data.AntivirusScanDate')
+
+    return ioc_context, ioc_readable, dbot_score, domain, ip_info, url_info, hash_info
+
+
+def request_for_ioc_enrichment():
+    """
+    Request for IOC enrichment
+    """
+    ioc_value = demisto.getArg('value')
+    request_url = 'public/v1/iocs/enrich/{}'.format(ioc_value)
+    ENRICHMENT_REQUEST_TIMEOUT_IN_SECONDS = 60 * 5
+    start_time = time.time()
+    while True:
+        if time.time() - start_time > ENRICHMENT_REQUEST_TIMEOUT_IN_SECONDS:
+            results_for_no_content('IOC Enrichment', 'Could not get any results. Reason: Request timeout.')
+            break
+        response = http_request('GET', request_url, json_response=True)
+        status = response.get('Status')
+        if status == 'Done':
+            ioc_context, ioc_readable, dbot_score, domain, ip_info, url_info, hash_info = ioc_enrichment_to_readable(response)
+
+            demisto.results(
+                {
+                    'Type': entryTypes['note'],
+                    'EntryContext': {
+                        'IntSights.Iocs(val.ID === obj.ID)': ioc_context,
+                        'DBotScore': dbot_score,
+                        'Domain': domain,
+                        'IP': ip_info,
+                        'URL': url_info,
+                        'File': hash_info
+                    },
+                    'Contents': response,
+                    'HumanReadable': tableToMarkdown('IOC Enrichment', ioc_readable),
+                    'ContentsFormat': formats['json']
+                }
+            )
+            break
+        elif status == 'Queued' or status == 'InProgress':
+            time.sleep(2)
```

Comment:
Could you please explain why this is needed?
Usage of `sleep` is strongly discouraged, using polling is usually the better option